### DISCUSSION TITLE
The storage of o3de.

### DISCUSSION THREAD

**Discussion by JKbasara** [o3de/o3de#1895] [General]
Just about two days ago, I downloaded a lumberyard of amazon. And I found the news "Linux Foundation to Form New Open 3D Foundation". It needs about 60G disk space. I just want to know is there any improvement about this cool opensource game engine to help optimize the storage use. I am a beginner for game engine or indiegame developer from China. I used UE4 , U3D and now I just learn some about godot. I really love the tiny engine. But the o3de (lumberyard) is evne larger than UE4 or cry engine. Is it means that it's not easy for a indiegame developer or hobbyist to use it?. Can anyone please tell me the basic concept about o3de. Thx first. And thank our opensource community to share that great 3d engine.

**Comment by Infurael** [comment]
The updated Gem system (O3DE's modular extension system) allows turning features on/off based on your projects needs. This will help reduce the size of what you need to compile and need to ship. Over time we hope to move more systems over to the new modular approach which will reduce the size of the currently required code. O3DE has also moved test and example projects into their own repositories to further reduce the download and storage sizes.

**Reply by JKbasara** [reply]
Thank you very much, I had know some basic concept. That sounds very great ~~~~

**Comment by AMZN-alexpete** [comment]
I'll add that the disk space requirements depend partially on what you do with O3DE.  If you intend to do engine development and rebuild the entire engine, tools _and_ make a game you're going to need a lot more disk space (100GB+) than if you use a pre-compiled version of the engine (coming soon) and build your game with scripting and just compiling your C++ game code if you have any.  In the end you will find that game assets (source + compiled per platform) end up taking the most space for a medium-to-large sized game - 200GB+

Here's some additional disk usage stats for the engine-development setup (including tests and the AutomatedTesting project)
- fresh clone of O3DE: ~ 7GB (3GB come from .git/lfs/objects)
- 3rdparty packages needed to compile the engine: ~12GB
- temporary files created during compiling: ~10GB
- binaries and debugging symbol files (editor.exe, editor.pdb etc): ~40GB


Compare the requirements for engine development above with the size of the installer version 
- O3DE engine sdk (includes binaries but no debugging symbols): ~10GB
- 3rdParty packages needed to compile the game: ~12GB (this will probably go down)

You can make this _much_ smaller ( ~ 2.5GB or 1 GB compressed ) if you remove all the parts of the SDK that are for compiling (libs, code, build files, 3rdparty) and the sample/test assets for distribution to team members who will never be compiling anything, but still need access to the tools (artists, game designers, etc).

**Reply by JKbasara** [reply]
Thank you very much and thank your detail explanation. That is so cool~~~
